tests/e2e/unified_test.go

```diff
@@ -273,11 +274,42 @@ func testFixturesInSeries(ctx context.Context, t *testing.T, testPause bool, can
 				}
 				create.RunCreateDeleteTest(h, opt)
 
-				for _, obj := range exportResources {
-					got := exportResource(h, obj)
-					if got != "" {
-						expectedPath := filepath.Join(fixture.SourceDir, "export.yaml")
-						h.CompareGoldenFile(expectedPath, string(got), IgnoreComments, ReplaceString(project.ProjectID, "example-project-id"))
+				if os.Getenv("GOLDEN_OBJECT_CHECKS") != "" {
+					for _, obj := range exportResources {
+						// Construct file path to save golden files
+						dir := "golden/"
+						pieces := strings.Split(t.Name(), "/")
+						var testName string
+						if len(pieces) > 0 {
+							testName = pieces[len(pieces)-1]
+						} else {
+							t.Errorf("failed to get test name")
+						}
+						// Golden test exported GCP object
+						got := exportResource(h, obj)
+						if got != "" {
+							expectedPath := filepath.Join(dir, fmt.Sprintf("export/_%v.yaml", testName))
```

Comment:
Shall we just write it to the fixture.SourceDir, so it lives with the test inputs ?

Reply:
I'm leaning towards we separate human-crafted files vs. generated-only files as it could distract the contributors. Though I do think having test data for the same resource to be centralized at one place makes sense. We could start with a more distinguishable name like `_generated_[testname].golden.yaml`. And in the long term, maybe creating a subfolder (we need to figure out a better regex to trigger individual test cases) called `generated` under each test case folder to hold all the generated test data.

Reply:
My intention was to put all generated files used for tests (i.e. vcr cassettes, golden files) together under a separate folder. I agree that the folder names and file names can be clearer so that we know they are generated files and used for vcr/golden testing.

Reply:
I like the underscore convention - that works for me!